lua/luasnip/extras/conditions/expand.lua

```diff
@@ -0,0 +1,14 @@
+local cond_obj = require("luasnip.extras.conditions")
+
+-- use the functions from show as basis and extend/overwrite functions specific for expand here
+local M = require("luasnip.extras.conditions.show")
+-----------------------
+-- PRESET CONDITIONS --
+-----------------------
+local function line_begin(line_to_cursor, matched_trigger)
+	-- +1 because `string.sub("abcd", 1, -2)` -> abc
+	return line_to_cursor:sub(1, -(#matched_trigger + 1)):match("^%s*$")
+end
+M.line_begin = cond_obj.make_condition(line_begin)
```

Comment:
I think this also modifies the `conditions.show`-table, `vim.tbl_extend` should do it

Reply:
no, `deepcopy` :facepalm:

Reply:
Should make no difference in this case (up to now the module only contains functions) I think (not entirely sure). But deepcopy definitely is more descriptive

Reply:
Yeah, `vim.tbl_extend` isn't optimal because the `copy` is a bit more hidden than with copy/deepcopy